core/src/main/java/com/airbnb/aerosolve/core/function/Function.java

```diff
@@ -26,7 +26,11 @@
 
   void resample(int newBins);
 
-  void smooth(double tolerance);
+  // for function not support smooth, just return 0
+  // for function support smooth, return errAndCoeff
+  float smooth(double tolerance);
+  // for function support smooth, return errAndCoeff percentage
+  float smoothByTolerancePercentage(double tolerancePercentage);
 
```

Comment:
Do these need to return float or double? Slightly odd when the tolerance is in double.